packages/volto/src/components/manage/Controlpanels/Controlpanel.jsx

```diff
@@ -113,6 +121,36 @@ class Controlpanel extends Component {
    * @returns {undefined}
    */
   UNSAFE_componentWillReceiveProps(nextProps) {
+    if (this.props.updateRequest.loading && nextProps.updateRequest.error) {
+      const message =
+        nextProps.updateRequest.error?.response?.body?.error?.message ||
+        nextProps.updateRequest.error?.response?.body?.message ||
+        nextProps.updateRequest.error?.response?.text ||
+        '';
+
+      const error =
```

Comment:
@wesleybl is p.restapi returning html as error? Where? It's wrong, and should be fixed, at least file an issue.

Reply:
@sneridagh In fact, I only saw restapi return html when we used

```
debug-exceptions = on
```

in buildout. To see: https://github.com/plone/plone.restapi/issues/1734

But perhaps this is an extreme case that is not even worth considering.

I just followed the way it is done when validating content forms:

https://github.com/plone/volto/blob/6f18663d012b1cdcaeb04d15a79f669973330109/packages/volto/src/components/manage/Add/Add.jsx#L178-L180

I also don't quite understand why this code is necessary. In my opinion it can be removed.

Perhaps @avoinea can explain why this is necessary, since he was the one who implemented the original code in:

https://github.com/plone/volto/commit/0782a50e6e8a8a9ada353ecd17228d7b423b77a0#diff-0356a7af508f860d87802dfa2b06d502aaed9cd60b130e57740806bd956be87e